### PR TITLE
Extract Factory Bot cops

### PR DESCRIPTION
**Pull request by ydah**
https://github.com/rubocop/rubocop-rspec/discussions/1440

### TODO
- [x] once the gem is published, remove the temp commit that adds rubocop-factory_bot to Gemfile
- [x] set the version constraint in gemspec for `rspec-factory_bot`
- [x] Creating a Repository for https://github.com/rubocop/rubocop-factory_bot
- [x] Extracted to https://github.com/rubocop/rubocop-factory_bot

All green ✅

Here's where to temporary extract it:.
- https://github.com/ydah/rubocop-factory_bot/pull/1
______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [-] Added tests.
- [-] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: rubocop-rspec.gemspec [hunk 1]
```diff
@@ -39,4 +39,5 @@ Gem::Specification.new do |spec|
 
   spec.add_runtime_dependency 'rubocop', '~> 1.33'
   spec.add_runtime_dependency 'rubocop-capybara', '~> 2.17'
+  spec.add_runtime_dependency 'rubocop-factory_bot'
```
Comment (pirj) [inline rubocop-rspec.gemspec:42]: Lock to a specific version? See #1567

**Review by pirj — COMMENTED** [review]
Looks great, thank you!
Just a few notes.

File: rubocop-rspec.gemspec (hunk 1 above)
Comment (ydah) [reply rubocop-rspec.gemspec]: We can't lock it down just yet, so I'm leaving it in the comments.

Comment (pirj) [reply rubocop-rspec.gemspec]: What would happen if you lock it here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@
 - Fix a false positive for `RSpec/PendingWithoutReason` when `skip` is passed a block inside example. ([@ydah], [@pirj])
 - Rename `RSpec/PendingBlockInsideExample` cop to `RSpec/SkipBlockInsideExample`. ([@pirj])
 - Deprecate `send_pattern`/`block_pattern`/`numblock_pattern` helpers in favour of using node pattern explicitly. ([@pirj], [@ydah])
+- Extract Factory Bot cops to a separate repository, [`rubocop-factory_bot`](https://github.com/rubocop/rubocop-factory_bot). The `rubocop-factory_bot` repository is a dependency of `rubocop-rspec` and the Capybara cops are aliased (`RSpec/FactoryBot/Foo` == `FactoryBot/Foo`) until v3.0 is released, so the change will be invisible to users until then. ([@pirj])
```
Comment (pirj) [inline CHANGELOG.md:17]:
```suggestion
- Extract Factory Bot cops to a separate repository, [`rubocop-factory_bot`](https://github.com/rubocop/rubocop-factory_bot). The `rubocop-factory_bot` repository is a dependency of `rubocop-rspec` and the Capybara cops are aliased (`RSpec/FactoryBot/Foo` == `FactoryBot/Foo`) until v3.0 is released, so the change will be invisible to users until then. ([@ydah])
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@
 - Fix a false positive for `RSpec/PendingWithoutReason` when `skip` is passed a block inside example. ([@ydah], [@pirj])
 - Rename `RSpec/PendingBlockInsideExample` cop to `RSpec/SkipBlockInsideExample`. ([@pirj])
 - Deprecate `send_pattern`/`block_pattern`/`numblock_pattern` helpers in favour of using node pattern explicitly. ([@pirj], [@ydah])
+- Extract Factory Bot cops to a separate repository. ([@ydah])
```
[the remark this answers is not in the document]
Comment (ydah) [reply CHANGELOG.md]: I got an error that such a version does not exist, but I may be forgetting something.

File: rubocop-rspec.gemspec (hunk 1 above)
Comment (ydah) [reply rubocop-rspec.gemspec]: I got an error that such a version does not exist, but I may be forgetting something.

**Comment by ydah** [comment]
@bquorning @Darhazer We would be happy if you could check here when you have time.

File: rubocop-rspec.gemspec [hunk 4]
```diff
@@ -39,4 +39,5 @@ Gem::Specification.new do |spec|
 
   spec.add_runtime_dependency 'rubocop', '~> 1.33'
   spec.add_runtime_dependency 'rubocop-capybara', '~> 2.17'
+  spec.add_runtime_dependency 'rubocop-factory_bot'# , '~> 2.20'
```
Comment (pirj) [inline rubocop-rspec.gemspec:42]: Just to try it out
```suggestion
  spec.add_runtime_dependency 'rubocop-factory_bot', '~> 2.20'
```

**Review by pirj — APPROVED** [review]
Thank you so much!

**Comment by ydah** [comment]
If you are authorized to create the following repositories, could you please create the following repositories?
https://github.com/rubocop/rubocop-factory_bot

**Comment by bquorning** [comment]
https://github.com/rubocop/rubocop-factory_bot has been created, and you should have received an email about getting access (as admin). I have given write permissions to the @rubocop/rubocop-rspec team too.

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,7 @@
 - Change `RSpec/ContainExactly` to ignore calls with no arguments, and change `RSpec/MatchArray` to ignore calls with an empty array literal argument. ([@ydah], [@bquorning])
 - Add new `RSpec/BeEmpty` cop. ([@ydah], [@bquorning])
 - Make `RSpec/MatchArray` and `RSpec/ContainExactly` pending. ([@ydah])
+- Extract Factory Bot cops to a separate repository, [`rubocop-factory_bot`](https://github.com/rubocop/rubocop-factory_bot). The `rubocop-factory_bot` repository is a dependency of `rubocop-rspec` and the Capybara cops are aliased (`RSpec/FactoryBot/Foo` == `FactoryBot/Foo`) until v3.0 is released, so the change will be invisible to users until then. ([@ydah])
```
Comment (pirj) [inline CHANGELOG.md:10]:
```suggestion
- Extract Factory Bot cops to a separate repository, [`rubocop-factory_bot`](https://github.com/rubocop/rubocop-factory_bot). The `rubocop-factory_bot` repository is a dependency of `rubocop-rspec` and the FactoryBot cops are aliased (`RSpec/FactoryBot/Foo` == `FactoryBot/Foo`) until v3.0 is released, so the change will be invisible to users until then. ([@ydah])
```

**Review by pirj — APPROVED** [review]
Thank you!

**Comment by bquorning** [comment]
@ydah Would you rebase on top of e.g. the `release` branch, and fix the last todo item?

**Comment by bquorning** [comment]
@ydah Feel free to also run `bundle exec rake cut_release:minor` and add the changes to this PR.

**Comment by ydah** [comment]
@bquorning Sorry for the late reply. I have updated this PR and ready for release ✅

**Comment by ydah** [comment]
@bquorning @pirj @Darhazer 
BTW, could you please invite me to join the rubocop-rspec team?
I can transfer issues between rubocop-rspec, rubocop_capybara and rubocop-factory_bot, which will make maintenance easier. So I hope you will consider this 🙏

**Comment by bquorning** [comment]
> BTW, could you please invite me to join the rubocop-rspec team?

Certain, and it’s about time 😅 An invitation has been sent.

**Review by bquorning — APPROVED** [review]

File: Gemfile [hunk 6]
```diff
@@ -15,3 +15,6 @@ gem 'yard'
 
 local_gemfile = 'Gemfile.local'
 eval_gemfile(local_gemfile) if File.exist?(local_gemfile)
+
+gem 'rubocop-factory_bot', git: 'https://github.com/ydah/rubocop-factory_bot',
+                           branch: 'extract-factory_bot'
```
[the remark this answers is not in the document]
Comment (pirj) [reply Gemfile]: Is this needed now when the gem is released? Can we point to rubygems?

[the remark this answers is not in the document]
Comment (pirj) [reply Gemfile]: My apologies, I’ve misunderstood this. I see no Gemfile changes, and gemspec defines dependencies 👍

**Comment by bquorning** [comment]
@pirj @ydah Is this ready for merge + release?

**Comment by ydah** [comment]
Thank you for the invitation to join the rubocop-rspec team!
Yes! Ready for release👍

**Comment by pirj** [comment]
Looks ready! 🙌

**Comment by bquorning** [comment]
And it’s official: https://rubygems.org/gems/rubocop-rspec/versions/2.22.0

Thank you @ydah 🎉

**Pull request merged**